File: app/src/main/java/com/example/rxjavaexamples/Car.java
package com.example.rxjavaexamples;

import java.util.Objects;

public class Car {
    private String model;
    private String year;

    public Car(String model, String year) {
        this.model = model;
        this.year = year;
    }

    public Car() {
    }

    @Override
    public String toString() {
        return "Car{" +
                "model='" + model + '\'' +
                ", year='" + year + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Car car = (Car) o;
        return model.equals(car.model) &&
                year.equals(car.year);
    }

    @Override
    public int hashCode() {
        return Objects.hash(model, year);
    }
}

File: app/src/main/java/com/example/rxjavaexamples/kotlinmodels/Car.kt
package com.example.rxjavaexamples.kotlinmodels

class Car(var model: String, var year: String) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Car

        if (model != other.model) return false
        if (year != other.year) return false

        return true
    }

    override fun hashCode(): Int {
        var result = model.hashCode();
        result = 31 * result + year.hashCode();
        return result;
    }

    override fun toString():String{
        return "Car: $model - $year";
    }
}